### PR TITLE
disable JSON_NUMERIC_CHECK by forcing mysql to not cast int/float types to strings

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | this is the point we need to check
| Fixed tickets | #2182

### CODE REVIEW

File: inc/dbutils.class.php [hunk 1]
```diff
@@ -704,7 +704,7 @@ public function getSonsOf($table, $IDf) {
 
       if (!is_array($sons)) {
          // IDs to be present in the final array
-         $sons[$IDf] = "$IDf";
```
Comment (orthagh) [inline inc/dbutils.class.php]: i need another pov for this one

**Comment by orthagh** [comment]
For information, after tests fixing, i think it's preferable to target 9.5 version to have a good beta/RC external revies

**Comment by orthagh** [comment]
For information, in case of PDO, we can have the same behavior with 
```
pdo->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, false);
pdo->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
```

File: inc/dbutils.class.php (hunk 1 above)
Comment (trasher) [reply inc/dbutils.class.php]: This does not change anything, in the end, we just use this array to the `exportArrayToDB` method, which I guess does not care about type hinting.

Comment (orthagh) [reply inc/dbutils.class.php]: is cache reset on update ?

Comment (trasher) [reply inc/dbutils.class.php]: Cache is updated at the end of the method

**Pull request merged**